tensorflow/compiler/tf2tensorrt/utils/trt_tensor_proxy.h

```diff
@@ -0,0 +1,599 @@
+/* Copyright 2018 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#ifndef TENSORFLOW_COMPILER_TF2TENSORRT_CONVERT_TRT_TENSOR_PROXY_H
+#define TENSORFLOW_COMPILER_TF2TENSORRT_CONVERT_TRT_TENSOR_PROXY_H
+
+#include <vector>
+#include <string>
+#include <cassert>
+#include <memory>
+
+#if GOOGLE_CUDA && GOOGLE_TENSORRT
+#include "third_party/tensorrt/NvInfer.h"
+namespace tensorflow {
+
+namespace tensorrt {
+
+namespace {
+  nvinfer1::Dims GetTestDims(const std::vector<int>& d) {
+    nvinfer1::Dims dims;
+    dims.nbDims = d.size();
+    for (int i = 0; i < d.size(); ++i) {
+      dims.d[i] = d[i];
+    }
+  }
+}
+// A fake ITensor implementation used to check whether the TF-TRT converter can
+// handle specific node. We only need shape and type information, and the
+// converter won't (and shouldn't) use this to build the TRT network.
```

Comment:
(1) Let's revise the comment:

SimpleITensor implements part of the ITensor interfaces to support the TF-TRT validator. As such, it only implements the interfaces that are related to shape and type information.

(2) can we remove the member functions that are not needed? For example, according to the comment, isNetworkInput shouldn't be needed. If that is not the case, we should fix the comment.

Reply:
Since we merged in the former test class FakeTensor, I think it might be best to leave the definitions of methods unused by the converter in case they're relevant to any tests (current or future).  Given that, I've updated the comment to say: `SimpleITensor implements part of the ITensor interfaces to support the TF-TRT validator, as well as some TF-TRT tests. The former use case only utilizes the interfaces related to shape and type information.`

Does this seem like a reasonable approach?

Reply:
I am fine with this.

Reply:
Sounds good, thanks!